src/components/SolidLoginForm.jsx

```diff
@@ -0,0 +1,67 @@
+import { getDefaultSession, fetch } from "@inrupt/solid-client-authn-browser";
+import {  getLiteral, getProfileAll, getThing } from "@inrupt/solid-client";
+import { FOAF } from "@inrupt/vocab-common-rdf";
+import { useState } from "react";
+
+function SolidLoginForm(props) {
+  const session = getDefaultSession();
+  let webId = session.info.webId
+  const [name, setName] = useState(undefined)
+  if (webId) {
+    getProfileAll(webId, { fetch: fetch }).then((dataSet) => {
+      let profile = dataSet.webIdProfile
+      const webIdThing = getThing(profile, webId)
+      setName(getLiteral(webIdThing, FOAF.name).value) 
+    });
+  }
+
+  /**
+   * Handling what should happen when the user trying to log in.
+   * @param {Event} event the event calling the EventListener  
+   */
+  function handleLogin(event) {
+    event.preventDefault();
+    props.onClick()
+    let idp = event.target[0].value;
+    session.login({
+      oidcIssuer: idp,
+      redirectUrl: new URL("/", window.location.href).toString(),
+      clientName: "Generic Data Viewer",
+    });
+  }
+
+  /**
+   * Handling what should happen whe the user logs out. 
+   * @param {Event} event the event calling the EventListener 
+   */
+  function handleLogout(event) {
+    event.preventDefault();
+    props.onClick()
+    session.logout();
+  }
+
+  if (!session.info.isLoggedIn) {
+    return (
+      <div className="login-form">
+        <form onSubmit={handleLogin} >
+        <input type="text" placeholder="Identity Provider..." defaultValue="https://inrupt.net" />
```

Comment:
The label is missing. It's not clear to the user what the content of the field should be.
Set the default value to `https://pod.playground.solidlab.be/`.

Reply:
I have adjusted both in [commit bf53b1a](https://github.com/SolidLabResearch/generic-data-viewer/pull/23/commits/bf53b1a96078c5feb5207239b0e86ef1b255e428)